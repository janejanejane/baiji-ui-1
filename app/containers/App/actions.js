
import {
  NAVIGATE_TO,
  ME_FROM_TOKEN,
  SET_LOGGED_IN_STATUS,
} from './constants';


export function navigateTo(url) {
  return {
    type: NAVIGATE_TO,
    url,
  };
}

export function meFromToken(token) {
  return {
    type: ME_FROM_TOKEN,
    token,
  };
}

export function setLoggedInStatus(status) {
  return {
    type: SET_LOGGED_IN_STATUS,
    status,
  };
}
