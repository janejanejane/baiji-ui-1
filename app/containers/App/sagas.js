import { call, put, takeLatest } from 'redux-saga/effects';

import request from 'utils/request';

import {
  ME_FROM_TOKEN,
} from './constants';

import {
  meRequestFailed,
  meRequestSuccess,
} from './actions';

export function* getMeFromToken(action) {
  // TODO: placeholder for request
  const url = `/me?token=${action.token}`;

  try {
    const me = yield call(request, url);
    // save new token to session storage
    sessionStorage.setItem('jwtToken', me.token);
    yield put(meRequestSuccess(me));
  } catch (err) {
    // delete token from session storage
    sessionStorage.removeItem('jwtToken');
    yield put(meRequestFailed(err.message));
  }
}

function* meData() {
  yield takeLatest(ME_FROM_TOKEN, getMeFromToken);
}

export default [
  meData,
];
