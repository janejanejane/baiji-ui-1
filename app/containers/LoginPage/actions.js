/*
 *
 * LoginPage actions
 *
 */

import {
  CHANGE_USERNAME,
  CHANGE_PASSWORD,
  LOGIN_SUCCESS,
  LOGIN_FAILED,
  SUBMIT_LOGIN_FORM,
} from './constants';

export function submitLoginForm({ username, password }) {
  return {
    type: SUBMIT_LOGIN_FORM,
    credentials: {
      username,
      password,
    },
  };
}

export function changeUsername(username) {
  return {
    type: CHANGE_USERNAME,
    username,
  };
}

export function changePassword(password) {
  return {
    type: CHANGE_PASSWORD,
    password,
  };
}

export function loginSuccess(data) {
  return {
    type: LOGIN_SUCCESS,
    data,
  };
}

export function loginFailed(error) {
  return {
    type: LOGIN_FAILED,
    error,
  };
}
