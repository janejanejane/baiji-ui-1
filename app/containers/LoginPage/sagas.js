import { take, takeLatest, call, put, select } from 'redux-saga/effects';

import request from 'utils/request';
import {
  changeUsername,
  changeUsernameError,
  loginSuccess,
  loginFailed,
} from './actions';
import {
  SUBMIT_LOGIN_FORM,
} from './constants';
import {
  makeSelectUsername,
  makeSelectPassword,
} from './selectors';

// Individual exports for testing
export function* getUsername() {
  const url = 'http://localhost:3000/username';

  try {
    const username = yield call(request, url);
    yield put(changeUsername(username));
  } catch (err) {
    yield put(changeUsernameError(err));
  }
}

export function* submitLoginForm() {
  const username = select(makeSelectUsername()); // get from selector
  const password = select(makeSelectPassword()); // get from selector
  const url = '/login';

  try {
    const loginData = yield call(request, url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: {
        username,
        password,
      },
    });
    yield put(loginSuccess(loginData));
  } catch (err) {
    yield put(loginFailed(err));
  }
}

export function* submitLoginTriggered() {
  yield takeLatest(SUBMIT_LOGIN_FORM, submitLoginForm);
}

// All sagas to be loaded
export default [
  submitLoginTriggered,
];
