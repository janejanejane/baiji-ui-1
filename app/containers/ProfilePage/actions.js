/*
 *
 * ProfilePage actions
 *
 */

import {
  LOAD_PROFILE,
  PROFILE_LOADED,
  PROFILE_LOAD_FAILED,
} from './constants';

export function loadProfile() {
  return {
    type: LOAD_PROFILE,
  };
}

export function loadProfileSuccess(user) {
  return {
    type: PROFILE_LOADED,
    user,
  };
}

export function loadProfileFailed(message) {
  return {
    type: PROFILE_LOAD_FAILED,
    message,
  };
}
