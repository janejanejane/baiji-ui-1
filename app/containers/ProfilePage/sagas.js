import { takeLatest, call, put } from 'redux-saga/effects';

import request from 'utils/request';

import {
  LOAD_PROFILE,
} from './constants';

import {
  loadProfileSuccess,
  loadProfileFailed,
} from './actions';

export function* loadProfileSaga(action) {
  const url = `/profile?id=${action.token}`;

  try {
    const profile = yield call(request, url);
    yield put(loadProfileSuccess(profile));
  } catch (err) {
    yield put(loadProfileFailed(err.message));
  }
}

// Individual exports for testing
export function* getProfile() {
  yield takeLatest(LOAD_PROFILE, loadProfileSaga);
}

// All sagas to be loaded
export default [
  getProfile,
];
