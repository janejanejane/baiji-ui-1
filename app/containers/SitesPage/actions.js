/*
 *
 * SitesPage actions
 *
 */

import {
  LOAD_SITES,
  LOAD_SITES_SUCCESS,
  LOAD_SITES_FAILED,
} from './constants';

export function loadSites() {
  return {
    type: LOAD_SITES,
  };
}

export function loadSitesSuccess(sites) {
  return {
    type: LOAD_SITES_SUCCESS,
    sites,
  };
}

export function loadSitesFailed(message) {
  return {
    type: LOAD_SITES_FAILED,
    message,
  };
}
