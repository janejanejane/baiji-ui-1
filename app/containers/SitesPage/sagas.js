import { takeLatest, call, put } from 'redux-saga/effects';

import request from 'utils/request';

import {
  LOAD_SITES,
} from './constants';

import {
  loadSitesSuccess,
  loadSitesFailed,
} from './actions';

// Individual exports for testing
export function* loadSitesSaga() {
  const url = '/sites';

  try {
    const sites = yield call(request, url);
    yield put(loadSitesSuccess(sites));
  } catch (err) {
    yield put(loadSitesFailed(err.message));
  }
}

export function* getSites() {
  yield takeLatest(LOAD_SITES, loadSitesSaga);
}

// All sagas to be loaded
export default [
  getSites,
];
