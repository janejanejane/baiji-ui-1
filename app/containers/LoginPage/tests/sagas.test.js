/**
 * Test  sagas
 */

/* eslint-disable redux-saga/yield-effects */
import { take, call, put, select } from 'redux-saga/effects';
import { loginSuccess, loginFailed } from '../actions';
import { submitLoginForm } from '../sagas';

describe('submitLoginForm Saga', () => {
  let submitLoginFormGenerator;

  beforeEach(() => {
    submitLoginFormGenerator = submitLoginForm();
    const testCall = submitLoginFormGenerator.next().value;

    expect(testCall).toMatchSnapshot();
  });

  it('should dispatch the loginSuccess action it request returns successfully', () => {
    const fixture = {
      firstName: 'firstName',
      lastName: 'lastName',
    };
    const callDescriptor = submitLoginFormGenerator.next(fixture).value;

    expect(callDescriptor).toEqual(put(loginSuccess(fixture)));
  });

  it('should dispatch the loginFailed action it request returns unsuccessfully', () => {
    const fixture = 'Error';
    const callDescriptor = submitLoginFormGenerator.throw(fixture).value;

    expect(callDescriptor).toEqual(put(loginFailed(fixture)));
  });
});
