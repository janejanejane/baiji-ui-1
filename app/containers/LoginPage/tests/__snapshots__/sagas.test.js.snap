exports[`submitLoginForm Saga should dispatch the loginFailed action it request returns unsuccessfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/login",
      Object {
        "body": Object {
          "password": Object {
            "@@redux-saga/IO": true,
            "SELECT": Object {
              "args": Array [],
              "selector": [Function],
            },
          },
          "username": Object {
            "@@redux-saga/IO": true,
            "SELECT": Object {
              "args": Array [],
              "selector": [Function],
            },
          },
        },
        "headers": Object {
          "Content-Type": "application/json",
        },
        "method": "POST",
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`submitLoginForm Saga should dispatch the loginSuccess action it request returns successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/login",
      Object {
        "body": Object {
          "password": Object {
            "@@redux-saga/IO": true,
            "SELECT": Object {
              "args": Array [],
              "selector": [Function],
            },
          },
          "username": Object {
            "@@redux-saga/IO": true,
            "SELECT": Object {
              "args": Array [],
              "selector": [Function],
            },
          },
        },
        "headers": Object {
          "Content-Type": "application/json",
        },
        "method": "POST",
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
